quickwit/quickwit-control-plane/src/indexing_plan.rs

```diff
@@ -272,67 +274,106 @@ impl From<IndexingTask> for SourceUid {
 ///   forward documents to the right indexers.
 /// - Ignore disabled sources, `CLI_INGEST_SOURCE_ID` and files sources (Quickwit is not aware of
 ///   the files locations and thus are ignored).
-pub(crate) fn build_indexing_plan(
-    source_configs: &HashMap<SourceUid, SourceConfig>,
+pub(crate) fn list_indexing_tasks(
     num_indexers: usize,
-) -> Vec<IndexingTask> {
-    let mut indexing_tasks: Vec<IndexingTask> = Vec::new();
-    for (source_uid, source_config) in source_configs
-        .iter()
+    model: &ControlPlaneModel,
+) -> Vec<LogicalIndexingTask> {
+    let mut indexing_tasks: Vec<LogicalIndexingTask> = Vec::new();
+    for (source_uid, source_config) in model
+        .get_source_configs()
         .sorted_by(|(left, _), (right, _)| left.cmp(right))
     {
         if !source_config.enabled {
             continue;
         }
-        if [CLI_INGEST_SOURCE_ID, INGEST_SOURCE_ID].contains(&source_config.source_id.as_str()) {
```

Comment:
I did not understand why the logic here was working off the source_id and not the source_type.
I moved that to a source type based logic. Is this risky?

Reply:
I don't know. It does not seem risky.